### ISSUE TITLE
Document how to install from source

### ISSUE THREAD

**Issue by Rodrigodd** [ahrm/sioyek#997]
From #505 I discover that just doing `make install` is not suppose to work.

I tried following the instructions in the issue (https://github.com/ahrm/sioyek/issues/505#issuecomment-1333534792), but I ended up with similar errors:

<details>

```shell
$ sudo ln /home/rodrigodd/repos/sioyek/build/sioyek /usr/bin/sioyek
$ which sioyek
/usr/bin/sioyek
$ sioyek
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
default_config_path: /usr/bin/prefs.config
Error: default_config_path: /usr/bin/prefs.config doesn't exist!
default_keys_path: /usr/bin/keys.config
Error: default_keys_path: /usr/bin/keys.config doesn't exist!
user_config_path: [ 0 ] /etc/xdg/sioyek/prefs_user.config
user_config_path: [ 1 ] /home/rodrigodd/.config/sioyek/prefs_user.config
user_config_path: [ 2 ] /home/rodrigodd/.local/share/Sioyek/prefs_user.config
user_config_path: [ 3 ] /home/rodrigodd/.config/sioyek/prefs_user.config
user_keys_path: [ 0 ] /etc/xdg/sioyek/keys_user.config
user_keys_path: [ 1 ] /home/rodrigodd/.config/sioyek/keys_user.config
user_keys_path: [ 2 ] /home/rodrigodd/.local/share/Sioyek/keys_user.config
user_keys_path: [ 3 ] /home/rodrigodd/.config/sioyek/keys_user.config
database_file_path: /home/rodrigodd/.local/share/Sioyek/test.db
local_database_file_path: /home/rodrigodd/.local/share/Sioyek/local.db
global_database_file_path: /home/rodrigodd/.local/share/Sioyek/shared.db
tutorial_path: /home/rodrigodd/.local/share/Sioyek/tutorial.pdf
last_opened_file_address_path: /home/rodrigodd/.local/share/Sioyek/last_document_path.txt
shader_path: /usr/bin/shaders
Error: shader_path: /usr/bin/shaders doesn't exist!
Creating shared memory block...
Shared memory created: this is the primary application.
Starting IPC server...
IPC server started.
```

</details>

Can instructions for installing from source be added to the `README.md`?

**Comment by ahrm** [comment]
The existing build instructions should work (except the development version uses qt 6.6 insetad of 5.12): https://github.com/ahrm/sioyek?tab=readme-ov-file#build-instructions .

**Comment by Rodrigodd** [comment]
Yes, the build instructions work, and I can run it from the build directory, but they do not explain how to install the editor system/user wide.

Running `./build_linux.sh` and then `make install` almost work, but I ran into #505, which was closed without a fix, and trying to symlink, as recommended, didn't work either, as described in the root comment.

**Comment by ahrm** [comment]
There is no "installing" just run the binary in the `build` directory.

**Comment by Rodrigodd** [comment]
Ah, ok. So installing from source is explicitly not supported?

**Comment by ahrm** [comment]
No.

**Comment by adigitoleo** [comment]
> they do not explain how to install the editor system/user wide

This depends on your system/package manager. You should probably consult its manuals instead. On linux, see also [install(1)](https://man7.org/linux/man-pages/man1/install.1.html).

**Comment by life00** [comment]
> No.

I am facing the same issue, except I have to build from source because my architecture (aarch64) is not supported. Please consider making it possible to properly install the application after its manually compiled.

> > they do not explain how to install the editor system/user wide
> 
> This depends on your system/package manager. You should probably consult its manuals instead. On linux, see also [install(1)](https://man7.org/linux/man-pages/man1/install.1.html).

I don't think this may help. I am not entirely sure how this work but I have an impression that these (relative) paths to runtime resources are hardcoded during the build process.

**Comment by life00** [comment]
I've tried building using `./build_and_release.sh` script (a bit modified) thinking that maybe it has runtime resources in more appropriate places, but no it appears it puts them in sioyek-release/usr/bin/ (as it is in the script) and it fails without them (especialy shaders). I don't really understand why does it do it?

**Comment by ahrm** [comment]
I honestly don't remember, I wrote that years ago for github actions and it still works so I took an if it ain't broke don't fix it approach, since debugging github actions is extremely annoying. However, note that for local builds (not meant for github actions) I use `build_linux.sh`.

**Comment by ahrm** [comment]
We have a `LINUX_STANDARD_PATHS` preprocessor definition which I think should help this problem.

**Comment by life00** [comment]
Compiling with this simple patch does the thing. @Rodrigodd only make sure to move the runtime resources to appropriate locations.

```diff
diff --git a/pdf_viewer/main.cpp b/pdf_viewer/main.cpp
index 61c4f45..75c908f 100644
--- a/pdf_viewer/main.cpp
+++ b/pdf_viewer/main.cpp
@@ -82,7 +82,7 @@
 #define MAX_PATH PATH_MAX
 #endif
 
-//#define LINUX_STANDARD_PATHS
+#define LINUX_STANDARD_PATHS
 
 
 std::wstring APPLICATION_NAME = L"sioyek";
```
```sh
cp prefs.config /etc/sioyek/
cp keys.config /etc/sioyek/
mkdir -p /usr/share/sioyek/
cp -r shaders/ /usr/share/sioyek/
cp sioyek /usr/bin/
cp tutorial.pdf /usr/share/sioyek/
```


I would say this issue may be considered to be resolved, however better documentation is preferred.
